plugins/inputs/socketstat/socketstat.go

```diff
@@ -0,0 +1,188 @@
+package socketstat
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// Socketstat is a telegraf plugin to gather indicators from established connections, using iproute2's  `ss` command.
+type Socketstat struct {
+	lister      socketLister
+	SocketProto []string
+	Timeout     internal.Duration
+}
+
+type socketLister func(proto string, Timeout internal.Duration) (*bytes.Buffer, error)
+
+const measurement = "socketstat"
+
+var defaultTimeout = internal.Duration{Duration: time.Second}
+
+// Description returns a short description of the plugin
+func (ss *Socketstat) Description() string {
+	return "Gather indicators from established connections, using iproute2's  `ss` command."
+}
+
+// SampleConfig returns sample configuration options
+func (ss *Socketstat) SampleConfig() string {
+	return `
+  ## ss can display information about tcp, udp, raw, unix, packet, dccp and sctp sockets
+  ## Specify here the types you want to gather
+  socket_proto = [ "tcp", "udp" ]
+  ## The default timeout of 1s for ss execution can be overridden here:
+  # timeout = "1s"
+`
+}
+
+// Gather gathers indicators from established connections
+func (ss *Socketstat) Gather(acc telegraf.Accumulator) error {
+	if len(ss.SocketProto) == 0 {
+		return nil
+	}
```

Comment:
You could check this in the `Init()` function mentioned above and fail early if no protocol is given. This reduces the amount of surprise... :-)

Reply:
Ok if you are not willing to introduce an `Init()` function, then either fail here or remove the `if` section completely as the for loop is not executed if `ss.SocketProto` is empty.

Reply:
done (moved to Init())